core/src/main/java/org/apache/accumulo/core/classloader/ClassLoaderUtil.java

```diff
@@ -75,11 +75,28 @@ static synchronized void resetContextFactoryForTests() {
 
   @SuppressWarnings("deprecation")
   public static ClassLoader getClassLoader(String context) {
-    if (context != null && !context.isEmpty()) {
-      return FACTORY.getClassLoader(context);
-    } else {
+    if (FACTORY == null) {
       return org.apache.accumulo.start.classloader.vfs.AccumuloVFSClassLoader.getClassLoader();
     }
```

Comment:
FACTORY isn't supposed to be null here when this is called, or below where a similar check is made. I'm not sure I understand this part of these changes.

If these methods can be called prior to the FACTORY being instantiated, then the FACTORY instantiation should be done lazily with a `Suppliers.memoize()`.

Reply:
I added this because I noticed that referencing FACTORY without calling `initContextFactory` would cause a NPE.  Looking at where this is called, I think the probability of it happening is low. I'm going to back this change out.

Reply:
Removed null checks in in 9272c0f